File: ios/CanvasDrawViewManager.m
//
//  DrawCanvas.m
//  Toddle
//
//  Created by Raj Jakasaniya on 07/07/20.
//

// CanvasViewManager.m


#import <React/RCTViewManager.h>
#import <React/RCTConvert.h>

@interface
RCT_EXTERN_MODULE(CanvasDrawViewManager, RCTViewManager)
RCT_EXPORT_VIEW_PROPERTY(strokeWidth , NSNumber)
RCT_EXPORT_VIEW_PROPERTY(strokeColor , NSString)
RCT_EXPORT_VIEW_PROPERTY(drawImagePath, NSString)
RCT_EXTERN_METHOD(clearCanvas:(nonnull NSNumber *)node)
RCT_EXTERN_METHOD(toggle:(nonnull NSNumber *)node)
RCT_EXTERN_METHOD(save:(nonnull NSNumber *)node)
RCT_EXTERN_METHOD(undo:(nonnull NSNumber *)node)
RCT_EXTERN_METHOD(redo:(nonnull NSNumber *)node)
RCT_EXPORT_VIEW_PROPERTY(onSaved, RCTDirectEventBlock)
RCT_EXPORT_VIEW_PROPERTY(onError, RCTDirectEventBlock)
@end

  

File: ios/CanvasDrawViewManager.swift
//
//  Canvas.swift
//  DrawCanvas
//
//  Created by Raj Jakasaniya on 07/07/20.
//  Copyright © 2020 Facebook. All rights reserved.
//

import Foundation
import UIKit


@objc(CanvasDrawViewManager)
class CanvasDrawViewManager: RCTViewManager {
  var canvas: CanvasDrawView!
  override func view() -> UIView! {
    canvas = CanvasDrawView()
    return canvas
  }
  
  override static func requiresMainQueueSetup() -> Bool {
    return true
  }
  
  @objc public func clearCanvas(_ node: NSNumber) {
    canvas.clearCanvas()
  }
  @objc public func redo(_ node: NSNumber) {
     canvas.redo()
   }
  @objc public func undo(_ node: NSNumber) {
     canvas.undo()
   }
  
  @objc public func save(_ node: NSNumber) {
    canvas.save()
  }
  
  @objc public func toggle(_ node: NSNumber) {
    canvas.toggle()
  }
}


